pkg/apis/core/validation/validation.go

```diff
@@ -2154,6 +2178,18 @@ func ValidatePersistentVolumeClaimSpec(spec *core.PersistentVolumeClaimSpec, fld
 		}
 	}
 
+	if spec.DataSourceRef2 != nil {
+		if !utilfeature.DefaultFeatureGate.Enabled(features.CrossNamespaceSourceProvisioning) {
+			allErrs = append(allErrs, field.Forbidden(fldPath.Child("dataSourceRef2"), "CrossNamespaceSourceProvisioning is disabled by feature-gate"))
+			spec.DataSourceRef2 = nil
+		} else {
+			allErrs = append(allErrs, validateDataSource2(spec.DataSourceRef2, fldPath.Child("dataSourceRef2"))...)
+		}
+		if spec.DataSource != nil {
+			allErrs = append(allErrs, field.Forbidden(fldPath.Child("dataSourceRef2"), "dataSourceRef2 reject. Because DataSource and DataSourceRef2 are both set."))
```

Comment:
I think `Invalid` may be a more fitting error for this. `Forbidden` seems to imply that it's some policy/permisisons issue.

Reply:
Please see https://github.com/kubernetes/kubernetes/pull/113186/commits/6db08e90f4e235396477549b0f7727b1abc62d68